test/runtests.jl

```diff
@@ -1,125 +1,311 @@
 using Impute
+using Tables
 using Test
 using DataFrames
 using RDatasets
 using Statistics
+using StatsBase
+
+import Impute:
+    Drop,
+    DropObs,
+    DropVars,
+    Interpolate,
+    Fill,
+    LOCF,
+    NOCB,
+    Context,
+    WeightedContext,
+    ImputeError
 
 @testset "Impute" begin
     a = Vector{Union{Float64, Missing}}(1.0:1.0:20.0)
     a[[2, 3, 7]] .= missing
     mask = map(!ismissing, a)
+    ctx = Context(; limit=0.2)
 
     @testset "Drop" begin
-        result = impute(a, :drop; limit=0.2)
-        expected = copy(a)
-        deleteat!(expected, [2, 3, 7])
+        @testset "DropObs" begin
+            result = impute(DropObs(; context=ctx), a)
+            expected = copy(a)
+            deleteat!(expected, [2, 3, 7])
 
-        @test result == expected
+            @test result == expected
+            @test result == Impute.dropobs(a; context=ctx)
+
+            a2 = copy(a)
+            Impute.dropobs!(a2; context=ctx)
+            @test a2 == expected
+        end
+        @testset "DropVars" begin
+            @testset "Matrix" begin
+                m = reshape(a, 5, 4)
+
+                result = impute(DropVars(; context=ctx), m)
+                expected = copy(m)[:, 2:4]
+
+                @test isequal(result, expected)
+                @test isequal(result, Impute.dropvars(m; context=ctx))
+                @test isequal(result', Impute.dropvars(m'; vardim=1, context=ctx))
+
+                Impute.dropvars!(m; context=ctx)
+                # The mutating test is broken because we need to making a copy of
+                # the original matrix
+                @test_broken isequal(m, expected)
+            end
+            @testset "DataFrame" begin
+                df = DataFrame(
+                    :sin => Vector{Union{Float64, Missing}}(sin.(1.0:1.0:20.0)),
+                    :cos => Vector{Union{Float64, Missing}}(sin.(1.0:1.0:20.0)),
+                )
+                df.sin[[2, 3, 7, 12, 19]] .= missing
+                df.cos[[4, 9]] .= missing
+
+                result = impute(DropVars(; context=ctx), df)
+                expected = df[[:cos]]
```

Comment:
I think this is likely the most robust way to access a column across DataFrames versions:
```suggestion
                expected = df.cos
```

Reply:
`df[[:cos]]` returns a `DataFrame`, not a vector. But better use `df[:, :cos]` or `select(df, :cos)` since `df[[:cos]]` is going to be deprecated very soon.

Reply:
Thanks. I didn't realize that was going away.

Reply:
df[:, :cos] also seems to return a vector, so only `select` does what I want.

Reply:
Sorry, I mean `df[:, [:cos]]`.

Reply:
Ahh, okay. Anyways, `select` worked. Thanks for the heads up.